frontend/src/features/modalSlice.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../app/store';
+
+export interface ModalState {
+  modalType: string | null;
+  modalProps?: Object;
+}
+
+const initialState: ModalState = {
+  modalType: null,
+  modalProps: {},
+};
+
+export const modalSlice = createSlice({
+  name: 'modal',
+  initialState,
+  reducers: {
+    // Use the PayloadAction type to declare the contents of `action.payload`
+    showModal: (state, action: PayloadAction<ModalState>) => {
+      state.modalType = action.payload.modalType;
+      state.modalProps = action.payload.modalProps
+        ? action.payload.modalProps
+        : {};
+    },
+
+    hideModal: (state) => {
+      state.modalType = initialState.modalType;
+      state.modalProps = initialState.modalProps;
```

Comment:
Would `state = initialState` work? My understanding is that redux makes everything immutable, so it would create a copy of the initial state.
You may want to wrap initial state inside `Object.freeze` just to be sure.

Reply:
It does not work. Because `state = initialState`, makes the local variable `state` point to `initialState` rather than change what `state` was initially pointing to. The redux state